### PR TITLE
Collections pagination

### PR DESCRIPTION
**Pull request by alxnddr**
### Description
Add pagination on the collection page
Screen Shot 2021-05-14 at 04 05 14

### CODE REVIEW

File: frontend/src/metabase/collections/components/PinnedItems.jsx [hunk 1]
```diff
@@ -30,12 +30,19 @@ const PinnedItem = ({ item, index, collection, onCopy, onMove }) => (
       onPin={() => item.setPinned(false)}
       onMove={onMove}
       onCopy={onCopy}
+      selectable={false}
```
Comment (kulyk) [inline frontend/src/metabase/collections/components/PinnedItems.jsx:33]: We actually want pinned items to be selectable for bulk actions. It was broken and fixed recently, here is the related issue: #15338

File: frontend/src/metabase/collections/components/ItemList.jsx [hunk 2]
```diff
@@ -77,18 +62,34 @@ export default function ItemList({
...
-              position: "relative",
-              height: ROW_HEIGHT * items.length,
-            }}
-          >
-            <VirtualizedList
-              items={items}
-              rowHeight={ROW_HEIGHT}
-              renderItem={renderListItem}
-              scrollElement={scrollElement}
-            />
```
Comment (kulyk) [inline frontend/src/metabase/collections/components/ItemList.jsx:91]: As we don't need to use `VirtualizedList` here anymore, we should also delete `scrollElement` from other places. I added it at #15715 to fix list virtualization, but now we can remove it completely. It's passed from the `CollectionLanding` down here through `CollectionContent`

**Review by kulyk — CHANGES_REQUESTED** [review]
Great work! I'm happy to see `CollectionContent` being a functional component now :) Thank for refactoring that. Everything looks good, there are just a couple of missing things.

Also, I see some of the tests related to collection permissions are failing. Feel free to ping me on those, I spend some time digging into them and I hope I can help

File: frontend/src/metabase/collections/components/PinnedItems.jsx (hunk 1 above)
Comment (alxnddr) [reply frontend/src/metabase/collections/components/PinnedItems.jsx:33]: Oh, good catch! I thought it was intended only for unpinned items and we just forgot to hide the checkbox. I'll fix that, thanks!

Comment (kulyk) [reply frontend/src/metabase/collections/components/PinnedItems.jsx:33]: I'm also worried that the repro test didn't fail 🤔 Can you give it a look please?

**Comment by alxnddr** [comment]
@kulyk thanks for the review, I fixed everything you mentioned, and the one failed due to a race condition spec.

Regarding pinned items selection, your PR went into `x.39.x` branch and it is not in the master yet. Since I did pretty much a lot of changes in the related code I decided to fix it right here. So after I merge this branch into master, I'll merge x.39.x and resolve these conflicts

**Review by daltojohnso — APPROVED** [review]
nice work!

File: frontend/src/metabase/collections/containers/CollectionContent.jsx [hunk 3]
```diff
@@ -1,200 +1,232 @@
...
+                  const showFilters =
+                    filter || unpinnedItems.length >= MIN_ITEMS_TO_SHOW_FILTERS;
+
+                  const unselected = unpinnedItems.filter(
+                    item => !getIsSelected(item),
+                  );
+                  const hasUnselected = unselected.length > 0;
+
+                  const handleSelectAll = () => {
+                    const pinnedUnseelcted = pinnedItems.filter(
```
Comment (daltojohnso) [inline frontend/src/metabase/collections/containers/CollectionContent.jsx:164]: nitpick -- `unseelcted`

Comment (alxnddr) [reply frontend/src/metabase/collections/containers/CollectionContent.jsx]: thanks, fixed that!

**Pull request merged**